### PR TITLE
Making FileProvider implementation optional

### PR DESCRIPTION
**Pull request by kkbhav**
# Overview
Previously, it was necessary to implement `ShareApplication` and pass the file provider authority from `MainApplication` file. This caused a lot of linking problems for people new to android. For this, I have added a `fileProvider` in the library itself. This will make implementing FileProvider optional and will help new users to get started quickly. Also, if people would like to use their `authority` for sharing they can implement the `ShareApplication` and get it running quickly without any other implications.

# Test Plan
We can test this out by using automatic installation and base64 images to share. This will use internal provider and then we can implement FileProvider in the main application and test it out. After implementing, the library will use a user-defined provider in place of ours. And in case the user-defined provider does not implement that path, the code will try it with our own provider.
I have tested this in an internal application and it works perfectly.

### CODE REVIEW

File: android/src/main/java/cl/json/RNShareFileProvider.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package cl.json;
+
+import android.support.v4.content.FileProvider;
+
+public class RNShareFileProvider extends FileProvider {
```
Comment (MateusAndrade) [inline android/src/main/java/cl/json/RNShareFileProvider.java]: This is correct? This class extends FileProvider but dont seems to be used on another place.

Comment (kkbhav) [reply android/src/main/java/cl/json/RNShareFileProvider.java]: It is used in the manifest. Android says we cannot declare multiple providers with the same name. So we had to implement a class which extends FileProvider and then declare provider in the manifest using this class

Comment (mikehardy) [reply android/src/main/java/cl/json/RNShareFileProvider.java]: That is true and what makes solving FileProvider configuration so difficult, I hadn't thought to simply extend to work around it, makes sense now that I think about it.

**Comment by mikehardy** [comment]
Sorry I haven't looked at this yet @kkbhav - I like this idea in theory - just need to look it over (or someone else can) - going to be very very busy until early next week unfortunately, but I do scan my PR list periodically so it won't be lost forever even if no one else gets to it

**Comment by kkbhav** [comment]
No problem.

**Comment by jgcmarins** [comment]
It looks good to me.
But I've been very busy too, co

**Comment by kkbhav** [comment]
Any updates?

**Comment by mikehardy** [comment]
Just got back from traveling all first part of week, today is PR review day, there is a nice stack of contributions in various projects - a good problem to have :muscle:  and will try to get everything approved and merged in where I can merge. And as mentioned before I love the idea here and appreciate the effort - thanks for your patience

**Comment by jgcmarins** [comment]
@kkbhav I always trust on your fixes and improvements \o/
If this one has no breaking changes, I think we are good to go

**Comment by mikehardy** [comment]
Oh - I will add I did not *run* it locally. Just close review of the code. Didn't have time to pull the branch etc sorry

**Comment by MateusAndrade** [comment]
> Oh - I will add I did not _run_ it locally. Just close review of the code. Didn't have time to pull the branch etc sorry

I can a try at it locally. If everything runs without problem we can proceed with the pr =D

**Comment by kkbhav** [comment]
@jgcmarins Thanks. It is always good to keep improving things. There are no breaking changes, this is just making `FileProvider` implementation optional. As @mikehardy said, this will help new Android users to get started quickly and advanced users to get all advanced features out of this library.

@mikehardy I just hope I have not irritated you 😅 . Thanks for doing the hard work and helping the community. I am pushing for this so that I can start using the npm version in place of a custom one.

@MateusAndrade Thanks man for giving it a run.

**Comment by kkbhav** [comment]
Please have a look at Readme.md. I have moved `FileProvider` details at the bottom of readme in the troubleshooting section as it became optional and only advanced users are gonna use it majorly.

**Comment by mikehardy** [comment]
> @mikehardy I just hope I have not irritated you 😅 .

Not a bit! thanks for the work

**Comment by jgcmarins** [comment]
Release?

**Comment by MateusAndrade** [comment]
> Release?

SURE! I pinned a issue #484 about the release. What do you think, on discussing the release process on this issue?

**Comment by mikehardy** [comment]
Yes! Even with problems, releasing is better than not releasing, for me philosophically. I don't have time to discuss now though, sorry :-(. Would have time in the next couple days to help with some urgent fix if needed though...

**Pull request merged**